### PR TITLE
[WIP] Containers Replicators Test refactoring for a faster execution

### PR DESCRIPTION
**Pull request by pavelzag**
{{pytest: cfme/tests/containers/test_containers_default_project_replicators.py -v --use-provider cm-env2 }}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 0073cb75795d9c7ea4fe02599b46616fee7dfbbc:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit a601f0bcad4665e4db4d05928aef34b8854ec137:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 2270493b967d59abc61d506b6b1f1fa23f877c5b:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

File: cfme/tests/containers/test_containers_default_project_replicators.py [hunk 1]
```diff
@@ -25,20 +25,33 @@
...
     navigate_to(Replicator, 'All')
     replicator_name = [r.name.text for r in list_tbl.rows()]
     for name in replicator_name:
         obj = Replicator(name, provider)
-        prop_val = obj.get_detail('Properties', ''.join(prop))
-        rel_val = obj.get_detail('Relationships', ''.join(rel))
-        assert prop_val
-        assert rel_val
+        assert obj.get_detail('Properties', ''.join(prop))
+        break
```
Comment (jaryn) [inline cfme/tests/containers/test_containers_default_project_replicators.py]: If there is the ``break`` here, why do we have the ``for`` above.

File: cfme/tests/containers/test_containers_default_project_replicators.py [hunk 2]
```diff
@@ -25,20 +25,33 @@
...
         """
     navigate_to(Replicator, 'All')
     replicator_name = [r.name.text for r in list_tbl.rows()]
     for name in replicator_name:
         obj = Replicator(name, provider)
-        prop_val = obj.get_detail('Properties', ''.join(prop))
-        rel_val = obj.get_detail('Relationships', ''.join(rel))
-        assert prop_val
-        assert rel_val
+        assert obj.get_detail('Properties', ''.join(prop))
```
Comment (jaryn) [inline cfme/tests/containers/test_containers_default_project_replicators.py]: We are still having the ```''.join(sometstring)``` here which is equivalent to the ```somestring``` itself

File: cfme/tests/containers/test_containers_default_project_replicators.py [hunk 3]
```diff
@@ -25,20 +25,33 @@
 # CMP - 9531
 
 
-@pytest.mark.parametrize(('prop', 'rel'), product(REPLICATORS_PROPERTIES_FIELDS,
-                                                  REPLICATORS_RELATIONSHIPS_FIELDS))
-def test_replicators_properties(provider, prop, rel):
+@pytest.mark.parametrize('prop', product(REPLICATORS_PROPERTIES_FIELDS))
```
Comment (jaryn) [inline cfme/tests/containers/test_containers_default_project_replicators.py]: Why there is the ```product``` here? ```product(X) == X```

**Comment by jaryn** [comment]
I fixed all the comments I had and improved the tests running time even more. I don't think we need to do ``      navigate_to(Replicator, 'All') `` for every Property or Relationship. We can navigate only once and then check all of it.

https://github.com/jaryn/integration_tests/tree/replicators_random_choice

File: cfme/tests/containers/test_containers_default_project_replicators.py (hunk 3 above)
Comment (psav) [inline cfme/tests/containers/test_containers_default_project_replicators.py]: ```product(X)``` doesn't necessarily == ```X```
as can be seen below
```
In [11]: product(REPLICATORS_PROPERTIES_FIELDS) == REPLICATORS_PROPERTIES_FIELDS
Out[11]: False
```
However, I know what you are trying to say, that essentially they are the same and yes in this case, we can remove ```product(REPLICATORS_PROPERTIES_FIELDS)``` usage and just replace it with ```REPLICATORS_PROPERTIES_FIELDS```.

File: cfme/tests/containers/test_containers_default_project_replicators.py (hunk 2 above)
Comment (psav) [inline cfme/tests/containers/test_containers_default_project_replicators.py]: Given the comment above, I agree that we can just replace this join with ```prop```

File: cfme/tests/containers/test_containers_default_project_replicators.py (hunk 1 above)
Comment (psav) [inline cfme/tests/containers/test_containers_default_project_replicators.py]: Agreed, why break, just use [0] if you are just going to test the first thing.

**Comment by dajoRH** [comment]
Lint report for commit f9d9f5795a912b7567e400c528b3dff21e44b573:

Lint OK! :cake: :punch: :cookie: .


**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 72b06284d9cea8c3682f4e00e5c629c0c2d8b092:

Lint OK! :cake: :punch: :cookie: .


**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by pavelzag** [comment]
Hey @psav, please have a look and let me know if there's something that needs to be done to get this merged

File: cfme/tests/containers/test_containers_default_project_replicators.py [hunk 4]
```diff
@@ -22,23 +22,32 @@
...
+        This test checks the properties fields of a Replicator
+        Steps :
+            * Goes to Containers --> Replicators
+             * Goes through one Replicator and  checks its Properties fields
+        """
+    navigate_to(Replicator, 'All')
+    replicator_names = [r.name.text for r in list_tbl.rows()]
+    assert 'router-1' in replicator_names
+    obj = Replicator(replicator_names[0], provider)
+    assert obj.get_detail('Properties', ''.join(prop))
```
Comment (nellyc) [inline cfme/tests/containers/test_containers_default_project_replicators.py]: polarion id missing

File: cfme/tests/containers/test_containers_default_project_replicators.py [hunk 5]
```diff
@@ -22,23 +22,32 @@
 REPLICATORS_RELATIONSHIPS_FIELDS = ['Containers Provider', 'Project', 'Pods', 'Nodes',
                                     'My Company Tags']
 
+
```
Comment (nellyc) [inline cfme/tests/containers/test_containers_default_project_replicators.py]: in general i dont understand what are we testing here - are we just making sure titles are ok? what about values?

Comment (pavelzag) [inline cfme/tests/containers/test_containers_default_project_replicators.py]: What do you mean? These are the titles of fields under Relationships

File: cfme/tests/containers/test_containers_default_project_replicators.py (hunk 4 above)
Comment (pavelzag) [inline cfme/tests/containers/test_containers_default_project_replicators.py]: No it's not # CMP - 9531

File: cfme/tests/containers/test_containers_default_project_replicators.py [hunk 6]
```diff
@@ -22,23 +22,32 @@
...
 # CMP - 9531
+@pytest.mark.parametrize('prop', product(REPLICATORS_PROPERTIES_FIELDS))
```
Comment (psav) [inline cfme/tests/containers/test_containers_default_project_replicators.py]: There is no need for product() here I think as we mentioned earlier

File: cfme/tests/containers/test_containers_default_project_replicators.py [hunk 7]
```diff
@@ -22,23 +22,32 @@
...
+    replicator_names = [r.name.text for r in list_tbl.rows()]
+    assert 'router-1' in replicator_names
+    obj = Replicator(replicator_names[0], provider)
+    assert obj.get_detail('Properties', ''.join(prop))
 
 
-@pytest.mark.parametrize(('prop', 'rel'), product(REPLICATORS_PROPERTIES_FIELDS,
-                                                  REPLICATORS_RELATIONSHIPS_FIELDS))
-def test_replicators_properties(provider, prop, rel):
+@pytest.mark.parametrize('rel', product(REPLICATORS_RELATIONSHIPS_FIELDS))
```
Comment (psav) [inline cfme/tests/containers/test_containers_default_project_replicators.py]: As above

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by pavelzag** [comment]
@psav 
A similar fix was merged with https://github.com/ManageIQ/integration_tests/pull/3856
I believe we can close this PR

**Comment by psav** [comment]
Closing as requested

**Pull request closed without merging**